### DISCUSSION TITLE
Updated post using putRecord. getRecord shows update as does getPosts however getPostThread not returning latest commit

### DISCUSSION THREAD

**Discussion by tmisra2004** [bluesky-social/atproto#3038] [Clarifications and "Why?"] [answered]
I updated a post using putRecord. The changes are being reflected in getRecord and getPosts but the changes are not being reflected in getPostThread. As a result of the change not being reflected in getPostThread, the updated post does not show on the BlueSky UI.

Post URI for reference is at://did:plc:z5xilekje47hjt24peedz7fj/app.bsky.feed.post/3lbbmmbsgee2p

**Comment by DavidBuchanan314** [comment]
I can see your post via getPostThread just fine, here: https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread?uri=at%3A%2F%2Ftmisra2004.bsky.social%2Fapp.bsky.feed.post%2F3lbbmmbsgee2p&depth=10 (also via the bsky.app web ui)

**Reply by tmisra2004** [reply]
HI the issue is not that the post isn't visible, it's that the latest commit is not showing in its place that was made via putRecord.

**Reply by DavidBuchanan314** [reply]
Ah yeah, I see. I think the short answer is "editing posts isn't supported yet", and the appview is caching the initial version. It's interesting if it shows up differently under getPosts, though - do you have an example of a feed that it's part of?

**Comment by yamarten** [comment, marked as answer]
[The Bluesky service prohibits updating posts](https://github.com/bluesky-social/atproto/pull/1385), so appview intentionally ignores it. However, it may be reflected in a short window, and [Skeets uses it to provide a post editing function.](https://bsky.app/profile/skeetsapp.com/post/3klv2zcovbu26)

putRecord does not fail because this is a specification unique to bsky post. PDS is expected to be lexicon agnostic, so putRecord succeeds, but appview, which is in charge of application logic, ignores it.

The old PDS included verification to update by [putRecord](https://github.com/bluesky-social/atproto/commit/6dfc899d995a0a7b0eb33ea1661e5c3660e38f90#diff-e35dfd17ddb08257c68e17800f771441e2e6298e8817dedaa1324442c42b0c1eL224) and [post recreating](https://github.com/bluesky-social/atproto/commit/6dfc899d995a0a7b0eb33ea1661e5c3660e38f90#diff-e35dfd17ddb08257c68e17800f771441e2e6298e8817dedaa1324442c42b0c1eL348), but this is a remnant of the time when PDS and appview were integrated and has now been removed.
